### PR TITLE
support helm ci test

### PR DESCRIPTION
**Pull request by zhu733756**
Signed-off-by: zhu733756 <zhu733756@kubesphere.io>

**What type of PR is this?**
Add helm ci test.
/kind feature

**What this PR does / why we need it**:
Enhance #3304 
**Which issue(s) this PR fixes**:

Part of #3300 

**Special notes for your reviewer**:
```
None
```

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

**Comment by gy95** [comment]
```
Which issue(s) this PR fixes:

Fixes #3300
```

can be replaced with `part of #3300` , and then the issue will not be closed once this patch be merged. :)

**Comment by zhu733756** [comment]
> ```
> Which issue(s) this PR fixes:
> 
> Fixes #3300
> ```
> 
> can be replaced with `part of #3300` , and then the issue will not be closed once this patch be merged. :)

 Seems great！Thanks

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -200,3 +200,61 @@ jobs:
...
+      
+      - name: Waiting for 10 seconds to keep cloudcore installed
+        uses: jakejarvis/wait-action@master
+        with:
+          time: '10s'
+
+      - name: Check cloudcore status
+        run: |
+          status=`kubectl get pods -n kubeedge | grep -i "cloudcore" |awk '{ print $3 }'`
+          if [ "$status" != "Running" ]; then
```
Comment (gy95) [inline .github/workflows/main.yaml:250]: whether we need to retry, just in case that in 10s, cloudcore still doesn't start.

Comment (zhu733756) [reply .github/workflows/main.yaml:250]: Maybe we need `--wait --timeout 30`,  this step will be removed. And the image have pushed to the kind cluster, there will be no blocks to set up.

**Comment by gy95** [comment]
Run cd build/helm && helm upgrade --install --wait --timeout 30 cloudcore ./cloudcore --namespace kubeedge --create-namespace -f ./cloudcore/values.yaml --set cloudCore.modules.cloudHub.advertiseAddress[0]=${NODE_IP} --set cloudCore.image.tag=${IMAGE_TAG}
Error: invalid argument "30" for "--timeout" flag: time: missing unit in duration 30

**Comment by zhu733756** [comment]
> Run cd build/helm && helm upgrade --install --wait --timeout 30 cloudcore ./cloudcore --namespace kubeedge --create-namespace -f ./cloudcore/values.yaml --set cloudCore.modules.cloudHub.advertiseAddress[0]=${NODE_IP} --set cloudCore.image.tag=${IMAGE_TAG} Error: invalid argument "30" for "--timeout" flag: time: missing unit in duration 30

Fixed.

File: .github/workflows/main.yaml (hunk 1 above)
Comment (fisherxu) [reply .github/workflows/main.yaml:244]: I wonder if we can write a script to do this and only call the script in github workflow?

Comment (zhu733756) [reply .github/workflows/main.yaml:244]: Reuse codes in keadm-e2e to implement the status checking and edge joining ？

File: .github/workflows/kind/kind.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+
```
Comment (zhu733756) [inline .github/workflows/kind/kind.yaml:1]: need to delete

File: .github/workflows/main.yaml (hunk 1 above)
Comment (fisherxu) [reply .github/workflows/main.yaml]: Yes, it will me more easy to maintain if in future we switch to other ci platform and can reuse the script.

Comment (zhu733756) [reply .github/workflows/main.yaml]: OK. I will have a test and make a re-pull request here. Currently, it got some bug.

Comment (fisherxu) [reply .github/workflows/main.yaml]: OK sounds good :)

**Comment by zhu733756** [comment]
Move to https://github.com/kubeedge/kubeedge/pull/3473

**Pull request closed without merging**